vmdb/app/assets/javascripts/miq_dhtmlxtoolbar.js

```diff
@@ -115,31 +115,34 @@ function miqToolbarOnClick(id){
 //	tb_hash = miq_toolbars.get(this.base.id);	// Use this line for toolbar v2.1
 		tb_hash = miq_toolbars[this.base.parentNode.id];	// Use this line for toolbar v3.0
 
-	eval("button = tb_hash['buttons']." + id);
-	if (typeof button != "undefined") {
-		if (typeof button.confirm != "undefined" && typeof button.popup == "undefined") {
+        if (!tb_hash['buttons'].hasOwnProperty(id)) {
+          return false;
+        }
+
+	button = tb_hash['buttons'][id];
+		if (button.hasOwnProperty("confirm") && !button.hasOwnProperty("popup")) {
 			if (!confirm(button.confirm)) return;
-		} else if (typeof button.confirm != "undefined" && typeof button.popup != "undefined") {
+		} else if (button.hasOwnProperty("confirm") && button.hasOwnProperty("popup")) {
 			// to open console in a new window
 			if (confirm(button.confirm)) {
-				if (typeof button.popup != "undefined" && button.popup) {					
-					if (typeof button.console_url != "undefined"){
+				if (button.popup != "undefined" && button.popup) {
+					if (button.hasOwnProperty("console_url")){
 						window.open(button.console_url);
 			}	}	}
 			return;			
-		} else if (typeof button.confirm == "undefined" && typeof button.popup != "undefined") {
+		} else if (!button.hasOwnProperty("confirm") && button.hasOwnProperty("popup")) {
 			// to open readonly report in a new window, doesnt have confirm message
-			if (typeof button.popup != "undefined" && button.popup) {
```

Comment:
I removed the `typeof button.popup` conditional because it's covered by the conditional on line 133.